app/src/main/java/org/mozilla/focus/locale/screen/LocaleFragmentCompose.kt

```diff
@@ -0,0 +1,173 @@
+        val listState = rememberLazyListState()
+        val coroutineScope = rememberCoroutineScope()
+        LaunchedEffect(0) {
+            coroutineScope.launch {
+                listState.scrollToItem(0)
+            }
+        }
+        val state = remember {
+            mutableStateOf("ro")
+        }
+        LanguagesList(
+            languageListItems = getFakeLanguages(),
+            state = state,
+            listState = listState
+        )
+    }
+}
+
+/**
+ * Displays a list of Languages in a listView
+ * @param languageListItems The list of Languages items to be displayed.
+ * @param state the current Selected Language
+ * @param listState scrolls to the current selected Language
+ */
+@Composable
+fun LanguagesList(
+    languageListItems: List<LanguageListItem>,
+    state: MutableState<String>,
+    listState: LazyListState
+) {
+    FocusTheme {
+        LazyColumn(
+            modifier = Modifier.background(colorResource(R.color.settings_background), shape = RectangleShape),
+            state = listState,
+            contentPadding = PaddingValues(horizontal = 12.dp),
+        ) {
+            items(languageListItems) { item ->
+                LanguageNameAndTagItem(
+                    language = item.language,
+                    isSelected = state.value == item.language.tag,
+                    onClick = item.onClick,
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun LanguageNameAndTagItem(
+    language: Language,
+    isSelected: Boolean,
+    onClick: (String) -> Unit
+) {
+    Row(
+        Modifier
+            .fillMaxWidth()
+            .wrapContentHeight(),
+        horizontalArrangement = Arrangement.Start,
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        LanguageRadioButton(
+            language = language,
+            isSelected = isSelected,
+            onClick = onClick
+        )
+        Spacer(modifier = Modifier.width(18.dp))
+        language.displayName?.let {
+            LanguageDisplayName(
+                language = language,
+                onClick = onClick
+            )
+        }
+    }
+}
+
+/**
+ * Displays a single language radiobutton
+ *
+ * @param language of the item
+ * @param isSelected check or uncheck the radioButton if the language is selected or not
+ * @param onClick Callback when the user taps on Language
+ */
+@Composable
+private fun LanguageRadioButton(
+    language: Language,
+    isSelected: Boolean,
+    onClick: (String) -> Unit
+) {
+    RadioButton(
+        selected = isSelected,
+        onClick = {
+            onClick(language.tag)
+        },
+    )
+}
+
+/**
+ * Displays a single language Text
+ *
+ * @param language of the item to be display in the textView
+ * @param onClick Callback when the user taps on Language text , the same like on the radioButton
+ */
+@Composable
+private fun LanguageDisplayName(language: Language, onClick: (String) -> Unit) {
+    if (language.displayName != null) {
```

Comment:
We already check for `language.displayName?` in Line 125 before calling `LanguageDisplayName`. We could simply just pass the `displayName` so we can avoid this check.

Reply:
I made the changes.